packages/codd/scripts/codd.py

```
@@ -343,6 +343,182 @@ def cmd_validate(root: Path, config: cc.CoddConfig) -> int:
+
+
+def _glob_prefix(pattern: str) -> str:
+    """glob パターンの先頭の固定（メタ文字以前の）プレフィックスを返す。"""
+    for index, char in enumerate(pattern):
+        if char in _GLOB_META:
+            return pattern[:index]
+    return pattern  # メタ文字なし（完全一致パターン）
+
+
+def _matches_scope_pattern(rel: str, pattern: str) -> bool:
+    """rel が scope glob にマッチするか（削除済みファイル向けの純粋パス判定）。
```

Ruff (0.15.18)

[warning] Docstring contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF002)